### PR TITLE
Keep trailing zeros when displaying Time and Duration

### PR DESCRIPTION
**Pull request by romainreignier**
Allow to have nicely aligned logs, like in roscpp

### CODE REVIEW

File: ros_message/src/time.rs [hunk 1]
```diff
@@ -97,21 +97,9 @@ impl Time {
     }
 }
 
-fn display_nanos(nanos: &str, f: &mut Formatter<'_>) -> fmt::Result {
-    let split_point = nanos.len() - 9;
-    let characters = nanos.chars();
-    let (left, right) = characters.as_str().split_at(split_point);
-    let right = right.trim_end_matches('0');
```
Comment (adnanademovic) [inline ros_message/src/time.rs:104]: Wouldn't just removing this line do the same trick?

My concern are perfectly valid but strange payloads, like `Duration { sec: -1, nsec: 1 }` and `Duration { sec: -1, nsec: -1 }` printing the same output in your modification of the code.

File: ros_message/src/time.rs [hunk 2]
```diff
@@ -150,14 +138,16 @@ impl Hash for Duration {
...
+        write!(
             f,
+            "{}{}.{:09}",
+            if self.sec.is_negative() || self.nsec.is_negative() {
+                "-"
+            } else {
+                ""
+            },
+            self.sec.abs(),
+            self.nsec.abs()
```
Comment (adnanademovic) [inline ros_message/src/time.rs:150]: This is the detail I was referencing in the other comment. With sec and nsec being -1, -1 we expect -1.000000001, while with nsec being 1 instead we'd expect -0.999999999.

I guess it's a failure of the test suite that I didn't add this in, as that was the reason for my more complex `display_nanos` function.

File: ros_message/src/time.rs (hunk 1 above)
Comment (adnanademovic) [reply ros_message/src/time.rs:104]: Of course, the if statement below is unnecessary then, as we're always triggering the else case.

**Review by adnanademovic — CHANGES_REQUESTED** [review]
Greatly appreciate the effort, and I fully agree with the desired functionality. If you'd return it to the way `display_nanos` worked but without trimming trailing zeros.

I don't want to adjust it myself to take away any of the credit you deserve for the change, but I can if you want me to.

**Comment by romainreignier** [comment]
I have done the requested changes and added a comment to explain the reason of the function.
I hope it's ok for you.

**Review by adnanademovic — APPROVED** [review]

**Comment by adnanademovic** [comment]
Thanks, this should work well now

**Pull request merged**